pkg/content/iterator/iterator.go

```diff
@@ -0,0 +1,73 @@
+package iterator
+
+import "github.com/readium/go-toolkit/pkg/content/element"
+
+// Iterates through a list of [Element] items asynchronously.
+// [hasNext] and [hasPrevious] refer to the last element computed by a previous call to any of both methods.
+// TODO: It's based on a kotlin iterator, maybe we can make this more of something for go?
```

Comment:
I didn't find much related to Go iterator, it looks like it's not very idiomatic. 🤷

Reply:
Yeah, I'm keeping this in sync with kotlin for now so I'm not going to change it as part of this PR at least